### PR TITLE
feat: new event when enclave calls fail

### PR DESCRIPTION
**Pull request by niklasad1**
As a UI/"fake mixer"/etc we would like to know which a particular call failed or succeeded. This PR is a step towards that goal however we need something better than `AccountId` for each event it doesn't help much.

### CODE REVIEW

File: frame/sgx-hello-world/src/lib.rs [hunk 1]
```diff
@@ -169,7 +169,8 @@ decl_event!(
 	pub enum Event<T> where AccountId = <T as frame_system::Trait>::AccountId {
 		EnclaveAdded(AccountId),
 		EnclaveRemoved(AccountId),
-		EnclaveCallDispatched(AccountId),
+		EnclaveCallSuccess(AccountId),
+		EnclaveCallFailure(AccountId),
```
Comment (niklasad1) [inline frame/sgx-hello-world/src/lib.rs]: Would be cool we could have a hash(account_id, call) then the `fake-mixer` could deduce which call got dispatched or failed.

Assuming that `call` is unique in this context.

The macro seems not to work for tuples tried `((AccountId, Vec<u8>))`

Comment (dvdplm) [reply frame/sgx-hello-world/src/lib.rs]: > hash(account_id, call) 

Would extrinsic hash work? If not, why not?

Comment (dvdplm) [reply frame/sgx-hello-world/src/lib.rs]: Not sure if it's a good idea or not, but would this work?

```
pub enum CallOutcome {
  WentSmoooooth,
  Naaah,
}
```
…and then do:
```
…
EnclaveCallDispatched(CallOutcome),
…
```

**Review by dvdplm — APPROVED** [review]
👍

File: frame/sgx-hello-world/src/lib.rs (hunk 1 above)
Comment (niklasad1) [reply frame/sgx-hello-world/src/lib.rs]: > Would extrinsic hash work? If not, why not?

Well, the extrinsic of the executed call in the enclave is not returned back to the `submit-extrinsic` and it depends on some stuff that changes during runtime such as nonce that we can not know about in advance. Even though the original call is constructed and known in `submit-extrinsic`. However, the encrypted payload will be intact and should be possible to subscribe to.

Thus, I implemented `T::Hasher::hash_of(&(&account_id, &encrypted_payload))` which should be possible to construct in the same way in `submit-extrinsic`

Ok?

EDIT: we could add nonce or something to protect against that the same calls being encrypted in the same way (FYI, I haven't checked how ecies_ed25519 encryption works)

**Comment by niklasad1** [comment]
> Not sure if it's a good idea or not, but would this work?

Works

**Pull request merged**